overlord/state/notices.go

```diff
@@ -107,15 +119,83 @@ func flattenUserID(userID *uint32) (uid uint32, isSet bool) {
 	return *userID, true
 }
 
+// Type represents a group of notices originating from a common source.
+func (n *Notice) Type() NoticeType {
+	return n.noticeType
+}
+
+// Key is a string that differentiates notices of the same type.
+func (n *Notice) Key() string {
+	return n.key
+}
+
+// FirstOccurred is the first time this notice occurs.
+func (n *Notice) FirstOccurred() time.Time {
+	return n.firstOccurred
+}
+
+// LastOccurred is the last time this notice occurred.
+func (n *Notice) LastOccurred() time.Time {
+	return n.lastOccurred
+}
+
+// LastRepeated is the time this notice was last "repeated".
+func (n *Notice) LastRepeated() time.Time {
+	return n.lastRepeated
+}
+
+// Occurrences is the number of times this notice has occurred.
+func (n *Notice) Occurrences() int {
+	return n.occurrences
+}
+
+// LastData is additional data captured from the last occurrence of this notice.
+func (n *Notice) LastData() map[string]string {
+	return n.lastData
+}
+
+// RepeatAfter is how long after this notice was last repeated should we allow
+// it to repeat.
+func (n *Notice) RepeatAfter() time.Duration {
+	return n.repeatAfter
+}
+
+// ExpireAfter is how long since this notice last occurred until we should drop it.
+func (n *Notice) ExpireAfter() time.Duration {
+	return n.expireAfter
+}
+
 // expired reports whether this notice has expired (relative to the given "now").
 func (n *Notice) expired(now time.Time) bool {
 	return n.lastOccurred.Add(n.expireAfter).Before(now)
 }
 
-// jsonNotice exists so we can control how a Notice is marshalled to JSON. It
-// needs to live in this package (rather than the API) because we save state
-// to disk as JSON.
-type jsonNotice struct {
+// GetRepeatCheckData unmarshals the previously stored repeat check data (from
+// AddNoticeOptions or RepeatCheck) into the value parameter
+func (n *Notice) GetRepeatCheckValue(value interface{}) error {
```

Comment:
Nit: discrepancy between name in comment (`GetRepeatCheckData`) and here ("value"). Also, `Get` implies a getter, whereas this kind of does parsing. Maybe `ParseRepeatCheckData` or `UnmarshalRepeatCheckData`?

Reply:
that's the pattern from customData though to call these things Get{,*}. This exists in snapd for example State.GetMaybeTimings(timings interface{}) error